### PR TITLE
fix consistency question for  tidbmonitor monitor  multiple cluster 

### PR DESCRIPTION
**Pull request by mikechengwei**
### What problem does this PR solve? 
#3531 
### What is changed and how does it work?
Every cluster generate a scrape job.
### Check List 

Tests

- Unit test
- E2E test
- Manual test (add detailed scripts or steps below)

Code changes

- Has Go code change

### CODE REVIEW

File: pkg/monitor/monitor/template.go [hunk 1]
```diff
@@ -126,13 +127,29 @@ func init() {
 }
 
 type MonitorConfigModel struct {
-	AlertmanagerURL    string
-	ReleaseNamespaces  []string
-	ReleaseTargetRegex *config.Regexp
-	EnableTLSCluster   bool
+	AlertmanagerURL  string
+	ClusterInfos     []ClusterRegexInfo
+	EnableTLSCluster bool
```
Comment (Yisaer) [inline pkg/monitor/monitor/template.go:132]: I think `EnableTLSCluster ` is better moved into `ClusterRegexInfo ` info.

File: pkg/monitor/monitor/template.go [hunk 2]
```diff
@@ -264,96 +270,114 @@ func buildAddressRelabelConfigByComponent(kind string) *config.RelabelConfig {
...
+		clusterTargetPattern, err := config.NewRegexp(cluster.Name)
+		if err != nil {
+			continue
+		}
+		nsTargetPattern, err := config.NewRegexp(cluster.Namespace)
+		if err != nil {
+			continue
+		}
+		scrapeJobs = append(scrapeJobs, &config.ScrapeConfig{
+			JobName:        jobName + "-" + strconv.Itoa(index),
```
Comment (Yisaer) [inline pkg/monitor/monitor/template.go:285]: I think using `clusterName` with `compoment` type as jobName is better.

File: pkg/monitor/monitor/template.go (hunk 1 above)
Comment (Yisaer) [reply pkg/monitor/monitor/template.go:132]: This could be revised in future.

File: pkg/monitor/monitor/template.go [hunk 3]
```diff
@@ -264,96 +270,114 @@ func buildAddressRelabelConfigByComponent(kind string) *config.RelabelConfig {
...
+	var scrapeJobs []*config.ScrapeConfig
+	for index, cluster := range cmodel.ClusterInfos {
+		clusterTargetPattern, err := config.NewRegexp(cluster.Name)
+		if err != nil {
+			continue
+		}
+		nsTargetPattern, err := config.NewRegexp(cluster.Namespace)
+		if err != nil {
+			continue
+		}
```
Comment (Yisaer) [inline pkg/monitor/monitor/template.go:283]: What about add a log when error happened here.

File: pkg/monitor/monitor/template_test.go [hunk 4]
```diff
@@ -517,10 +535,8 @@ scrape_configs:
     action: replace
 `
 	model := &MonitorConfigModel{
-		ReleaseTargetRegex: &target,
-		ReleaseNamespaces: []string{
-			"ns1",
-			"ns2",
+		ClusterInfos: []ClusterRegexInfo{
+			{Name: "target", Namespace: "ns1"},
```
Comment (Yisaer) [inline pkg/monitor/monitor/template_test.go:539]: It's better to verify multi tidbcluster here.

**Review by Yisaer — COMMENTED** [review]
Rest LGTM

**Comment by mikechengwei** [comment]
Manual Test Screenshot：
image

image

File: pkg/monitor/monitor/template_test.go (hunk 4 above)
Comment (mikechengwei) [reply pkg/monitor/monitor/template_test.go:479]: ok,i take time to make up.

**Review by Yisaer — COMMENTED** [review]
Alomost LGTM, plz add the unit test for multi tidbclusters case.

File: pkg/monitor/monitor/template.go [hunk 5]
```diff
@@ -264,96 +269,113 @@ func buildAddressRelabelConfigByComponent(kind string) *config.RelabelConfig {
...
+					Action:      config.RelabelReplace,
+					TargetLabel: "instance",
 				},
-				Action:      config.RelabelReplace,
-				TargetLabel: "cluster",
 			},
-		},
+		}
+		if cmodel.EnableTLSCluster {
+			scrapeconfig.HTTPClientConfig.TLSConfig = config.TLSConfig{
```
Comment (DanielZhangQD) [inline pkg/monitor/monitor/template.go:368]: No need to set certs for lightning, please check the detail in #3532.

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Comment by DanielZhangQD** [comment]
@Yisaer PTAL again, thanks!

**Review by Yisaer — APPROVED** [review]
LGTM

**Pull request merged**